controllers/profile.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+var bodyParser = require('body-parser');
+var session = require('express-session');
+var bcrypt = require('bcrypt');
+var db = require('../database.js');
+var config = require('../config.js');
+var MySQLStore = require('express-mysql-session')(session);
+var sessionStore = new MySQLStore(config.dbOptions);
+
+function requireAuthentication(req, res, next) {
+
+	if (!req.isAuthenticated()) {
+		return res.redirect('/login');
+	}
+
+	// The user is logged in.F
+	// Allow heir request to continue.
+	next();
+}
+
+
+module.exports = function(app) {
+
+	app.get('/profile', function(req, res, next) {
+		var sessionUser= req.session.user.username;
+		db.knex('users')
+			.where({username: sessionUser })
+				.select('email')
+				.then(function(result){
+					if (!result || !result[0]){ // not found
+						res.send('Strange... Very strange...');
+					}
+					//hence db call is async we have to render inside .then
+					res.render('profile', {username : sessionUser, email:result[0].email});
+				})
+		
+	});
+
+
+	app.post('/profile', function(req, res) {
+
+			// the following variables take the
+			// data username and password from the form
+			var usernameReq = req.body.username;
+			var passwordReq = req.body.password;
+			var emailReq = req.body.email;
+			// knex compares the information from the form
+			// with the information in the user table
+			// and update email if the user and login is right
+			db.knex('users')
+			.where({username: usernameReq })
+				.select('password')
+				.then(function(result){
+					if (!result || !result[0]){ // not found
+						res.send('Strange... Very strange...');
+					}
+					var hash = result[0].password;
+					if (bcrypt.compareSync(passwordReq, hash)){
```

Comment:
Should avoid using synchronous methods for hashing passwords. This blocks the event loop and prevents the server from doing anything else while it waits for the hashing to complete.